src/pynwb/misc.py

```diff
@@ -149,7 +150,8 @@ class Units(DynamicTable):
         {'name': 'electrode_group', 'description': 'the electrode group that each spike unit came from'},
         {'name': 'waveform_mean', 'description': 'the spike waveform mean for each spike unit'},
         {'name': 'waveform_sd', 'description': 'the spike waveform standard deviation for each spike unit'},
-        {'name': 'waveforms', 'description': waveforms_desc, 'index': 2}
+        {'name': 'waveforms', 'description': waveforms_desc, 'index': 2,
+         'class': MeasurementData, 'unit': 'volts', 'conversion': 1., 'offset': 0.}
```

Comment:
@rly I had assumed this logic would propagate to https://github.com/catalystneuro/hdmf/blob/dev/src/hdmf/common/table.py#L473-L480 and allow the values to be specified at `__init__` of the `UnitsTable` in this fashion, but it's not recognizing that this subclass of `VectorData` has any additional arguments - any ideas?

How would you recommend passing these values (conversion + offset and maybe units) both for the initiation of the `UnitsTable` (including cases where no units have waveforms added later), as well as manual user specification of these values (possibly during `add_unit`)?

Reply:
`allow_extra=True` needs to be added to the docval of `DynamicTable.add_column`, i.e., 
```python
            {'name': 'col_cls', 'type': type, 'default': VectorData,
             'doc': ('class to use to represent the column data. If table=True, this field is ignored and a '
                     'DynamicTableRegion object is used. If enum=True, this field is ignored and a EnumData '
                     'object is used.')}, 
            allow_extra=True)
    def add_column(self, **kwargs):  # noqa: C901
```

The tests still fail with that because conversion=1. is being passed but a VectorData is expected based on the spec